include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -243,6 +244,32 @@ inline ConstMemoryInfo AllocatorImpl<T>::GetInfo() const {
   return ConstMemoryInfo{out};
 }
 
+template <typename T>
+inline std::unordered_map<std::string, std::string> AllocatorImpl<T>::GetStats() const {
+  AllocatorWithDefaultOptions allocator;
+  char* raw_stats = nullptr;
+  ThrowOnError(GetApi().AllocatorGetStats(this->p_, allocator, &raw_stats));
+  std::unordered_map<std::string, std::string> stats;
+  if (raw_stats == nullptr) {
+    return stats;
+  }
```

Comment:
This should not happen. Users will not check both for Status code and empty map. This is not a good API.